src/linux/LinuxKeyboard.cpp

```diff
@@ -353,16 +237,78 @@ void LinuxKeyboard::capture()
 	}
 }
 
+//-------------------------------------------------------------------//
+void LinuxKeyboard::_handleKeyPress( XEvent& event )
+{
+	XKeyEvent& e = (XKeyEvent&)event;
+	static std::vector<char> buf(32);
+	KeySym keySym;
+	unsigned int character = 0;
+	int bytes = 0;
+	bool haveChar = true;
+
+	haveChar = !XFilterEvent(&event, None);
+
+	if (xic)
+	{
+		Status status;
+		do
+		{
+			bytes = Xutf8LookupString(xic, &e, &buf[0], buf.size()-1, &keySym, &status);
+			buf[bytes] = '\0';
+
+			if (status == XBufferOverflow)
+				buf.resize(buf.size() * 2);

```

Comment:
This is actually the stock reallocation strategy of std::vector when doing a push_back... It would be better to implement this as "reserving" X char at initialization of the vector, and letting the STL work as intended